runner/app/live/streamer/process_guardian.py

```diff
@@ -158,18 +158,28 @@ def _compute_current_state(self) -> str:
         # Special case: stream not running
         current_time = time.time()
         input = self.status.input_status
-        time_since_last_input = current_time - (input.last_input_time or 0)
+        start_time = max(self.process.start_time, self.status.start_time)
+        time_since_last_input = current_time - (input.last_input_time or start_time)
 
         if not self.streamer.is_stream_running():
+            # give it 3s `DEGRADED_INPUT` grace period after shutdown
+            is_offline = time_since_last_input > 3 or not input.last_input_time
+            return PipelineState.OFFLINE if is_offline else PipelineState.DEGRADED_INPUT
+
+        # Special case: stream shutdown after inactivity
+        if time_since_last_input > 60:
+            if self.streamer.trigger_stop_stream():
+                logging.info(
+                    f"Shutting down streamer. Flagging DEGRADED_INPUT state during shutdown: time_since_last_input={time_since_last_input:.1f}s"
+                )
             return (
-                PipelineState.OFFLINE
-                if time_since_last_input > 3  # 3s grace period after shutdown
-                else PipelineState.DEGRADED_INPUT
+                PipelineState.DEGRADED_INPUT
+                if time_since_last_input < 90
+                else PipelineState.ERROR  # Not shutting down after 30s, declare ERROR
```

Comment:
should this say 90s instead of 30s?

Reply:
it was referring to the 30s after the trigger above, so 90s total. I elaborated the comment further so it's clearer